tmnt-org/tmnt
# Add sign-in via google functionality

- Add google sign-in button to login page
- Add backend logic to process response

![Screenshot 2023-03-09 at 1 13 31 PM](https://user-images.githubusercontent.com/47332368/224160888-2105a412-7ce3-47e7-9975-119df4fea131.png)

Testing (and all works as expected)
- Sign in using traditional sign-in
- SIgn in using google sign-in - the first time (i.e. no DB record), and subsequent entry (with DB record)
- Sign up with new and known username

Though I would like someone else to test this as well to see if it works as expected for them.

Note that clicking `create board` after sign-in throws an error in the backend, but is out of scope in the current PR to fix:

```
/Users/nattiechan/projects/Scratch-Project/server/controllers/boardController.js:22
      message: { err: "boardController.createBoard" + err.message },
                                                      ^

ReferenceError: err is not defined
    at boardController.createBoard (/Users/nattiechan/projects/Scratch-Project/server/controllers/boardController.js:22:55)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

Node.js v18.12.1
```
